src/libcrun/cgroup-systemd.c

```diff
@@ -1487,7 +1490,43 @@ append_resources (sd_bus_message *m,
                   return ret;
               }
           }
+
+        if (! is_update && has_bpf_fs ())
+          {
+            const char *property_name = "BPFProgram";
+
+            if (! property_missing_p (missing_properties, property_name))
+              {
+                cleanup_free struct bpf_program *program = NULL;
+
+                program = create_dev_bpf (resources->devices, resources->devices_len, err);
+                if (LIKELY (program != NULL))
+                  {
+                    cleanup_free char *path = NULL;
+
+                    mkdir(CRUN_BPF_DIR, 0700); // Best effort.
+
+                    ret = append_paths (&path, err, CRUN_BPF_DIR, id, NULL);
```

Comment:
it really stinks that systemd requires us to install a program instead of just reading a file, but definitely an improvement compared to the current status.

What we have in place is enough to make sure it is deleted when the container is deleted?